zookeeper-contrib/zookeeper-contrib-monitoring/check_zookeeper.py3

```diff
@@ -0,0 +1,404 @@
+            if m is not None:
+                result['zk_ephemerals_count'] = int(m.group(1))
+                continue
+
+            m = re.match('Approximate data size: (\d+)', line)
+            if m is not None:
+                result['zk_approximate_data_size'] = int(m.group(1))
+                continue
+
+            m = re.match('Open file descriptor count: (\d+)', line)
+            if m is not None:
+                result['zk_open_file_descriptor_count'] = int(m.group(1))
+                continue
+
+            m = re.match('Max file descriptor count: (\d+)', line)
+            if m is not None:
+                result['zk_max_file_descriptor_count'] = int(m.group(1))
+                continue
+
+            m = re.match('Zxid: (0x[0-9a-fA-F]+)', line)
+            if m is not None:
+                result['zk_zxid']         = m.group(1)
+                result['zk_zxid_counter'] = int(m.group(1), 16) & int('0xffffffff', 16) # lower 32 bits
+                result['zk_zxid_epoch']   = int(m.group(1), 16) >>32 # high 32 bits
+                continue
+
+            m = re.match('Proposal sizes last/min/max: (\d+)/(\d+)/(\d+)', line)
+            if m is not None:
+                result['zk_last_proposal_size'] = int(m.group(1))
+                result['zk_min_proposal_size'] = int(m.group(2))
+                result['zk_max_proposal_size'] = int(m.group(3))
+                continue
+
+        return result
+
+    def _parse_line(self, line):
+        try:
+            key, value = map(str.strip, line.split('\t'))
+        except ValueError:
+            raise ValueError('Found invalid line: %s' % line)
+
+        if not key:
+            raise ValueError('The key is mandatory and should not be empty')
+
+        for typ in [int, float]:
+            try:
+                value = typ(value)
+                break
+            except (TypeError, ValueError):
+                pass
+
+        return key, value
+
+def main():
+    opts, args = parse_cli()
+    cluster_stats = get_cluster_stats(opts.servers)
+    if opts.output is None:
+        dump_stats(cluster_stats)
+        return 0
+
+    handler = create_handler(opts.output)
+    if handler is None:
+        log.error('undefined handler: %s' % opts.output)
+        sys.exit(1)
+
+    return handler.analyze(opts, cluster_stats)
+
+def create_handler(name):
+    """ Return an instance of a platform specific analyzer """
+    try:
+        return globals()['%sHandler' % name.capitalize()]()
+    except KeyError:
+        return None
+
+def get_all_handlers():
+    """ Get a list containing all the platform specific analyzers """
+    return [NagiosHandler, CactiHandler, GangliaHandler]
+
+def dump_stats(cluster_stats):
+    """ Dump cluster statistics in an user friendly format """
+    for server, stats in cluster_stats.items():
+        print ('Server:'), server
+
+        for key, value in stats.items():
+            print ("%30s") % key, ' ', value
```

Comment:
With python3.8, this line cannot pass the compiler. Maybe I'm wrong since I don't know python.
My chatgpt tell me it should be like `print("{:<30} {}".format(key, value)) ` and it works well